relay-system/src/controller.rs

```diff
@@ -55,11 +58,22 @@ pub use actix::actors::signal::{Signal, SignalType};
 ///     Ok(())
 /// }).unwrap();
 /// ```
+
+type ShutdownReceiver = watch::Receiver<Option<Shutdown>>; // Find a good name for this
+type ShutdownSender = watch::Sender<Option<Shutdown>>; // Check of that is really needed
+                                                       // Only used once but gives a nice symetry
+
+/// TODO
 pub struct Controller {
     /// Configured timeout for graceful shutdowns.
     timeout: Duration,
     /// Subscribed actors for the shutdown message.
     subscribers: Vec<Recipient<Shutdown>>,
+
+    // Hand this out to actors that subscribe to us
```

Comment:
nit: write doc comments in 3rd person, use sentence-ending punctuation even for single sentences.
```suggestion
    /// Handed out to actors who wish to subcribe to the [`Shutdown`] message.
```